src/TLuaInterpreter.cpp

```diff
@@ -3989,18 +3992,62 @@ int TLuaInterpreter::connectToServer( lua_State *L )
 
     if( ! lua_isnumber( L, 2 ) )
     {
-        lua_pushstring( L, "connectToServer: wrong argument type" );
-        lua_error( L );
-        return 1;
+        lua_pushstring( L, QString(" connectToServer: argument #2 expects type int, %1 was given.").arg( lua_typename(L,2) ).toUtf8().constData() );
+        return lua_error( L );
     }
     else
     {
         port = lua_tonumber( L, 2 );
+
+        if( port > 65535 || port < 0 )
+        {
+            lua_pushstring( L, " connectToServer: argument #2 is invalid, a valid port number is required.");
+            return lua_error(L);
+        }
     }
+
+    // Optional argument to save this new connection to disk for this profile.
+    if( !lua_isnoneornil(L,3) && !lua_isboolean(L,3) ) {
+        lua_pushstring( L, QString(" connectToServer: argument #3 expects boolean type, %1 was given.").arg( lua_typename(L,3) ).toUtf8().constData() );
+        return lua_error(L);
+    }
+    if( ! lua_isnoneornil(L,3) && lua_isboolean(L,3)) {
+        saveToProfile = lua_toboolean(L,3);
+        if( saveToProfile ) {
+            qDebug() << "TLuaInterpreter::connectToProfile(" << url.c_str() << ", " << port << ", " << saveToProfile << ")";
+
+            const QString filePath("%1/.config/mudlet/profiles/%2/%3");
+            QFile urlFile( filePath.arg(QDir::homePath(), pHost->getName(), "url") );
+            QFile portFile( filePath.arg(QDir::homePath(), pHost->getName(), "port") );
+
+            if( urlFile.open(QIODevice::WriteOnly) ) {
+                QString newURL = url.c_str();
+                QDataStream ofsu(&urlFile);
+                ofsu << newURL;
+                urlFile.close();
+            } else {
+                lua_pushnil(L);
+                lua_pushstring( L, QString(" connectToServer: Save has Failed - %1").arg(urlFile.errorString()).toUtf8().constData() );
```

Comment:
Yeah, there are not too many samples of that new _run-time_ error type messages in the existing code but we (__I__) will be using it in the future after @vadi2 pointed out (__firmly__ 😜 🔨) to me, the rationale for using it...!